File: kaggle_environments/envs/kore_fleets/starter_bots/java/kore/Cell.java
package kore;

import java.util.Optional;

public class Cell {

    public final Point position;
    public double kore;
    public String shipyardId;
    public String fleetId;
    public final Board board;
    
    public Cell(Point position, double kore, String shipyardId, String fleetId, Board board) {
        this.position = position;
        this.kore = kore;
        this.shipyardId = shipyardId;
        this.fleetId = fleetId;
        this.board = board;
    }

    public Cell cloneToBoard(Board board) {
        return new Cell(this.position, this.kore, this.shipyardId, this.fleetId, board);
    }

    public Optional<Fleet> fleet() {
        if (this.board.fleets.containsKey(this.fleetId)) {
            return Optional.of(this.board.fleets.get(this.fleetId));
        }
        return Optional.empty();
    }


    public Optional<Shipyard> shipyard() {
        if (this.board.shipyards.containsKey(this.shipyardId)) {
            return Optional.of(this.board.shipyards.get(this.shipyardId));
        }
        return Optional.empty();
    }

    public Cell neighbor(Point offset) {
        Point next = this.position.translate(offset, this.board.size);
        return this.board.getCellAtPosition(next);
    }


    public Cell north() {
        return this.neighbor(Direction.NORTH);
    }

    public Cell south() {
        return this.neighbor(((Point)Direction.SOUTH));
    }

    public Cell east() {
        return this.neighbor(Direction.EAST);
    }

    public Cell west() {
        return this.neighbor(Direction.WEST);
    }
}

File: kaggle_environments/envs/kore_fleets/starter_bots/ts/kore/Cell.ts
import {Board} from "./Board";
import {Direction} from "./Direction";
import {Fleet} from "./Fleet";
import {Point} from "./Point";
import {Shipyard} from "./Shipyard";

export class Cell {

    public readonly position: Point;
    public kore: number;
    public shipyardId: string;
    public fleetId: string;
    public readonly board: Board;
    
    public constructor(position: Point, kore: number, shipyardId: string, fleetId: string, board: Board) {
        this.position = position;
        this.kore = kore;
        this.shipyardId = shipyardId;
        this.fleetId = fleetId;
        this.board = board;
    }

    public cloneToBoard(board: Board): Cell {
        return new Cell(this.position, this.kore, this.shipyardId, this.fleetId, board);
    }

    public get fleet(): Fleet | undefined {
        if (this.board.fleets.has(this.fleetId)) {
            return this.board.fleets.get(this.fleetId);
        }
        return undefined;
    }


    public get shipyard(): Shipyard | undefined {
        if (this.board.shipyards.has(this.shipyardId)) {
            return this.board.shipyards.get(this.shipyardId);
        }
        return undefined;
    }

    public neighbor(offset: Point): Cell {
        const next = this.position.translate(offset, this.board.size);
        return this.board.getCellAtPosition(next);
    }


    public north(): Cell {
        return this.neighbor(Direction.NORTH);
    }

    public south(): Cell {
        return this.neighbor((Direction.SOUTH));
    }

    public east(): Cell {
        return this.neighbor(Direction.EAST);
    }

    public west(): Cell {
        return this.neighbor(Direction.WEST);
    }
}
